src/main.rs

```diff
@@ -495,29 +491,20 @@ fn start_vmm(cmd_arguments: ArgMatches) -> Result<Option<String>, Error> {
         SeccompAction::Trap
     };
 
-    // See https://github.com/rust-lang/libc/issues/716 why we can't get the details from siginfo_t
-    if seccomp_action == SeccompAction::Trap {
-        thread::Builder::new()
-            .name("seccomp_signal_handler".to_string())
-            .spawn(move || {
-                for si in SignalsInfo::<WithRawSiginfo>::new(&[SIGSYS])
-                    .unwrap()
-                    .forever()
-                {
-                    /* SYS_SECCOMP */
-                    if si.si_code == 1 {
-                        eprint!(
-                            "\n==== seccomp violation ====\n\
-                            Try running with `strace -ff` to identify the cause and open an issue: \
-                            https://github.com/cloud-hypervisor/cloud-hypervisor/issues/new\n"
-                        );
-
-                        signal_hook::low_level::emulate_default_handler(SIGSYS).unwrap();
-                    }
-                }
-            })
-            .unwrap();
+    // SAFETY: We only using signal_hook for managing signals and only execute signal
+    // handler safe functions (writing to stderr) and manipulating signals.
+    unsafe {
```

Comment:
Did you remove the condition
```rust
if seccomp_action == SeccompAction::Trap {
```
on purpose?

Reply:
Yes, I dropped it by accident, in practice the other situation is rare, running binary using new syscalls on an older kernel.